libraries/lib-nyquist-effects/NyquistBase.cpp

```diff
@@ -2744,19 +2744,7 @@ void NyquistBase::OSCallback()
       nyx_continue();
    }
 
-   // LLL:  STF figured out that yielding while the effect is being applied
-   //       produces an EXTREME slowdown.  It appears that yielding is not
-   //       really necessary on Linux and Windows.
-   //
-   //       However, on the Mac, the spinning cursor appears during longer
-   //       Nyquist processing and that may cause the user to think Audacity
-   //       has crashed or hung.  In addition, yielding or not on the Mac
-   //       doesn't seem to make much of a difference in execution time.
-   //
-   //       So, yielding on the Mac only...
-#if defined(__WXMAC__)
-   wxYieldIfNeeded();
-#endif
+   YieldIfNeededHook::Call();
```

Comment:
Wouldn't `virtual void NyquistBase::OSYield() = 0` be simpler solution in that case?

Reply:
I thought about that, but it feels even more wrong to use a virtual member function, meanwhile the call to yield in fact just needs a stateless function.
I tried something else, similarly to what we did for `PlatformCompatibility`:
``` 
lib-nyquist-effects/NyquistRoutines.h
src/effects/nyquist/NyquistRoutines.cpp
```
with `void YieldIfNeeded()` declared in the DLL and defined in the executable. But that fails on Windows, because building a DLL actually implies linkage, and then the definition isn't found.
It worked for `PlatformCompatibility` because there I could place `PlatformCompatibilityWX.cpp` in the DLL itself, since `wxStandardPaths` is in wxBase, which doesn't turn a library pink.
If this inspires an alternative solution to you, I'd gladly hear it.